### PR TITLE
gnuradio3_8: 3.8.2.0 -> 3.8.3.0

### PR DESCRIPTION
**Pull request by eduardosm**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by eduardosm** [comment]
cc @doronbehar

**Comment by SuperSandro2000** [comment]
Maybe this helps you https://github.com/gnuradio/gnuradio/issues/4278 https://github.com/drowe67/codec2/issues/60 . I think their toolchain is not supporting layouts like nix has quite well.

**Comment by eduardosm** [comment]
Fixed by adding `-DLIBCODEC2_FOUND=TRUE` and `-DLIBGSM_FOUND=TRUE`. Maybe this will work for gr3.9 too.

File: pkgs/applications/radio/gnuradio/3.8.nix [hunk 1]
```diff
@@ -244,9 +244,11 @@ stdenv.mkDerivation rec {
     # detected properly. NOTE: qradiolink needs libcodec2 to be detected in
     # order to build, see https://github.com/qradiolink/qradiolink/issues/67
     ++ lib.optionals (hasFeature "gr-vocoder" features) [
+      "-DLIBCODEC2_FOUND=TRUE"
```
Comment (doronbehar) [inline pkgs/applications/radio/gnuradio/3.8.nix:247]:
```suggestion
      # From some reason we have to add all of these flags, the issue is reported
      # upstream: https://github.com/gnuradio/gnuradio/issues/4278
      "-DLIBCODEC2_FOUND=TRUE"
```

**Review by doronbehar — CHANGES_REQUESTED** [review]

**Review by doronbehar — APPROVED** [review]

**Comment by doronbehar** [comment]
Can be merged if ofborg is green, I'm going to sleep. Ping me if it's not merged by tomorrow :).

**Comment by eduardosm** [comment]
@doronbehar Remember to merge :)

**Comment by risicle** [comment]
WFM linux x86_64

File: pkgs/applications/radio/gnuradio/3.8.nix [hunk 2]
```diff
@@ -262,23 +267,4 @@ stdenv.mkDerivation rec {
       ${removeReferencesTo}/bin/remove-references-to -t ${python} $out/lib/cmake/gnuradio/GnuradioConfig.cmake
     ''
   ;
-  patches = [
-    # Don't install python referencing files if python support is disabled.
-    # See: https://github.com/gnuradio/gnuradio/pull/3839
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/applications/radio/gnuradio/3.8.nix:268]: Please remove fetchpatch from inputs.

Comment (eduardosm) [reply pkgs/applications/radio/gnuradio/3.8.nix:268]: https://github.com/NixOS/nixpkgs/pull/118496

**Pull request merged**